dev-academy/develop-smart-contract/02-develop.md

```diff
@@ -176,211 +146,483 @@ pub fn execute(
+* The new entry is defined with the received `description` and `priority` attributes. The `status` of the new entry is set to `ToDo` by default. Notice that `priority` is an optional parameter. If not provided, the default value will be `None`.
+* The function finally saves the new entry to the `LIST` with the matching `id` and returns a `Response` with the relevant attributes. 
 
 ```rust
-//msg.rs
-#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
-#[serde(rename_all = "snake_case")]
-pub enum ExecuteMsg {
-    CreatePot {
-        /// target_addr will receive tokens when token amount threshold is met.
-        target_addr: String,
-        /// threshold is the token amount for releasing tokens.
-        threshold: Uint128,
-    },
-    /// Receive forwards received cw20 tokens to an execution logic
-    Receive(Cw20ReceiveMsg),
+//contract.rs
+pub fn create_new_entry(deps: DepsMut, info: MessageInfo, description: String, priority: Option<Priority>) -> Result<Response, ContractError> {
+    let owner = CONFIG.load(deps.storage)?.owner;
+    if info.sender != owner {
+        return Err(ContractError::Unauthorized {});
+    }
+    let id = ENTRY_SEQ.update::<_, cosmwasm_std::StdError>(deps.storage, |id| {
+        Ok(id.add(1))
+    })?;
+    let new_entry = Entry {
+        id,
+        description,
+        priority: priority.unwrap_or(Priority::None),
+        status: Status::ToDo
+    };
+    LIST.save(deps.storage, id, &new_entry)?;
+    Ok(Response::new().add_attribute("method", "create_new_entry")
+        .add_attribute("new_entry_id", id.to_string()))
 }
-
 ```
-
-Now that we have updated `ExecuteMsg`, we must create `execute_create_pot` which is called by the contract's `Execute` function.
+The function `update_entry()` handles the update of an existing entry on the list. 
+* Before continuing with the new update, the function checks if the message sender is the owner of the contract. If not, it returns an error and the update fails to be performed.
+* The entry with the matching `id` is loaded from the `LIST`.
+* Sharing the same id, an updated version of the entry is defined with the received values for `description`, `status` and `priority`. These are optional parameters and if any one of them is not provided, the function defaults back to the corresponding value from the entry loaded.
+* The function saves the updated entry to the `LIST` with the matching `id` and returns a `Response` with the relevant attributes. 
 
 ```rust
-// contract.rs
-pub fn execute_create_pot(
-    deps: DepsMut,
-    info: MessageInfo,
-    target_addr: String,
-    threshold: Uint128,
-) -> Result<Response, ContractError> {
-    // owner authentication
-    let config = CONFIG.load(deps.storage)?;
-    if config.owner != info.sender {
+//contract.rs
+pub fn update_entry(deps: DepsMut, info: MessageInfo, id: u64, description: Option<String>, status: Option<Status>, priority: Option<Priority>) -> Result<Response, ContractError> {
```

Comment:
Maybe you can explain the steps above in code comments? So the logic and implementation walks together